[bitnami/grafana-loki] Implement query-scheduler configuration

bcarlock-emerge opened the pull request:
### Description of the change

Implementation of the grafana-loki query-scheduler and corresponding changes to query-frontend and querier.
https://grafana.com/docs/loki/latest/operations/scalability/#separate-query-scheduler
https://github.com/grafana/loki/blob/main/production/ksonnet/loki/query-scheduler.libsonnet

### Benefits

Allows query-frontend & querier to scale more gracefully, increase performance. 
"The Query frontend has an in-memory queue that can be moved out into a separate process similar to the [Cortex Query Scheduler](https://cortexmetrics.io/docs/operations/scaling-query-frontend/#query-scheduler). This allows running multiple query frontends." https://grafana.com/docs/loki/latest/operations/scalability/#separate-query-scheduler

### Possible drawbacks

N/A- as far as I'm aware

### Applicable issues

N/A

### Additional information

Tests have been run and compared against a running manual configuration with all expected values returned. 

### Checklist

<!-- [Place an '[X]' (no spaces) in all applicable fields. Please remove unrelated fields.] -->

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

fmulero commented on bitnami/grafana-loki/Chart.yaml:
```diff
@@ -44,4 +44,4 @@ name: grafana-loki
   - https://github.com/bitnami/bitnami-docker-grafana-loki
   - https://github.com/grafana/loki/
-version: 2.1.2
+version: 2.2.2
```
I think there is a typo here:
```suggestion
version: 2.2.0
```

fmulero commented on bitnami/grafana-loki/templates/query-scheduler/deployment.yaml:
```diff
@@ -0,0 +1,165 @@
+  {{- if .Values.queryScheduler.updateStrategy }}
+  strategy: {{- toYaml .Values.queryScheduler.updateStrategy | nindent 4 }}
+  {{- end }}
+  revisionHistoryLimit: 2
```
`revisionHistoryLimit` is not used in other deployments. We should keep consistency between all the deployments in this chart. Could you remove it?

fmulero commented on bitnami/grafana-loki/templates/query-scheduler/deployment.yaml:
```diff
@@ -0,0 +1,165 @@
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  minReadySeconds: 10
```
Is there any requirement or reason to set `minReadySeconds`? If there is a reason to use it, could make sense to create a value that allow users change it

bcarlock-emerge replied:
I'm honestly not sure, it was generated by Tanka. I'll add it as a value.

fmulero commented on bitnami/grafana-loki/values.yaml:
```diff
@@ -200,9 +200,10 @@ loki:
             validity: 24h
           {{- end }}
-
+    {{- if not .Values.queryScheduler.enabled }}
```
This is affecting to all grafana-loki componente, not only the querier. Is it expected?

fmulero reviewed (changes requested):
Great @bcarlock-emerge, nice work! Could you have a look to the comments?

fmulero commented:
Hi @bcarlock-emerge, there are too many changes related to other charts, not sure if you had any problem during merge. Could you resolve conflicts and check if the changes were done intentionally?

bcarlock-emerge commented:
> Hi @bcarlock-emerge, there are too many changes related to other charts, not sure if you had any problem during merge. Could you resolve conflicts and check if the changes were done intentionally?

Sorry for the delay, I've addressed the issues with other charts.
